src/zulip.rs

```diff
@@ -476,6 +479,206 @@ async fn whoami_cmd(
+                ));
+            };
+            username
+        }
+    };
+
+    Ok(format!("{zulip_username}'s GitHub profile is [{github_username}](https://github.com/{github_username})."))
+}
+
+/// Tries to find a Zulip username from a GitHub username.
+async fn lookup_zulip_username(ctx: &Context, gh_username: &str) -> anyhow::Result<String> {
+    async fn lookup_from_zulip(ctx: &Context, gh_username: &str) -> anyhow::Result<Option<String>> {
+        let username_lowercase = gh_username.to_lowercase();
+        let users = get_zulip_users(ctx.github.raw()).await?;
+        Ok(users
+            .into_iter()
+            .find(|user| {
+                user.get_github_username()
+                    .map(|u| u.to_lowercase())
+                    .as_deref()
+                    == Some(username_lowercase.as_str())
+            })
+            .map(|u| u.name))
+    }
+
+    async fn lookup_from_team(ctx: &Context, gh_username: &str) -> anyhow::Result<Option<String>> {
+        let people = people(&ctx.github).await?.people;
+
+        // Lookup the person in the team DB
+        let Some(person) = people.get(gh_username).or_else(|| {
+            let username_lowercase = gh_username.to_lowercase();
+            people
+                .keys()
+                .find(|key| key.to_lowercase() == username_lowercase)
+                .and_then(|key| people.get(key))
+        }) else {
+            return Ok(None);
+        };
+
+        let Some(zulip_id) = to_zulip_id(&ctx.github, person.github_id).await? else {
+            return Ok(None);
+        };
+        let Ok(zulip_user) = get_zulip_user(&ctx.github.raw(), zulip_id).await else {
+            return Ok(None);
+        };
+        Ok(Some(zulip_user.name))
+    }
+
+    let zulip_username = match lookup_from_team(ctx, gh_username).await? {
+        Some(username) => username,
+        None => match lookup_from_zulip(ctx, gh_username).await? {
+            Some(username) => username,
+            None => {
+                return Ok(format!(
+                    "No Zulip account found for GitHub username `{gh_username}`."
+                ))
+            }
+        },
+    };
+    Ok(format!(
+        "The GitHub user `{gh_username}` has the following Zulip account: @**{zulip_username}**"
+    ))
+}
+
+#[derive(Clone, serde::Deserialize, Debug, PartialEq, Eq)]
+pub(crate) struct ProfileValue {
+    value: String,
+}
+
+/// A single Zulip user
+#[derive(Clone, serde::Deserialize, Debug, PartialEq, Eq)]
+pub(crate) struct ZulipUser {
+    pub(crate) user_id: u64,
+    #[serde(rename = "full_name")]
+    pub(crate) name: String,
+    #[serde(default)]
+    pub(crate) profile_data: HashMap<String, ProfileValue>,
+}
+
+impl ZulipUser {
+    // The GitHub profile data key is 3873
+    pub(crate) fn get_github_username(&self) -> Option<&str> {
+        self.profile_data.get("3873").map(|v| v.value.as_str())
```

Comment:
How did you figure this out? Is there any documentation about it? Or stability at all?

Reply:
I have no idea, to be honest :/ I just tried to load the data from Rust Zulip, and this ID is used for GitHub user profiles. I guess that someone created it for our Zulip (https://zulip.com/help/custom-profile-fields), and it got assigned some ID. It's probably not portable across Zulip instances, but hopefully it should stay stable on the Rust Zulip.

Reply:
Hum, I didn't knew it was linked to custom-profile-fields, I always though it was "native" integration.

Would be good to mention that in the function, with the fact that the ID is r-l Zulip instance specific.